crates/ty_python_semantic/src/types.rs

```diff
@@ -3631,7 +3629,6 @@ impl<'db> Type<'db> {
                 // TODO: Consider merging this signature with the one in the previous match clause,
                 // since the previous one is just this signature with the `self` parameters
                 // removed.
-                let not_none = Type::none(db).negate(db);
```

Comment:
I think you may want to apply the same change to the overload just above, right? It's slightly harder to write a test for, because you might need to call `__get__` manually? Let me know if you need help.

Reply:
thanks, yes, you're correct that we emitted false positive diagnostics for `object.__str__.__get__(object(), None)()` as well as `object().__str__()` 😆

I added a regression test!

Reply:
> `object.__str__.__get__(object(), None)()`

Just imagine how happy users will be that this *finally* works correctly.